src/app/modules/auth/classic-auth/classic-auth.service.ts

```diff
@@ -160,6 +165,36 @@ export class ClassicAuthService {
     }
   }
 
+  async resendActivationEmail (classicAuthActivateResendPayloadDto :ClassicAuthActivateResendPayloadDto) {
+    const message = {
+      message: this.i18n.t('auth.mail.activation', {
+        lang: 'en',
+      })
+    };
+    try {
+      const user = await this.classicAuthRepository.findOne({where: {email: classicAuthActivateResendPayloadDto.email}});
+
+      if(!user) {
+        return message;
+      }
+
+      const activationCode = v4 ();
+      await this.classicAuthRepository.update({email: user.email},{
+        activation_code: activationCode,
+      });
+
+      await this.mailerService.sendActivationEmail (
+        classicAuthActivateResendPayloadDto.email,
+        this.generateActivationLink(activationCode),
+        user.name
+      );
+
+      return message;
+    } catch (e) {
+      throw e;
```

Comment:
Did you test this? Api must return a readable error. In case of error, user must see what happens